proto/minder/v1/minder.proto

```diff
@@ -1232,6 +1232,18 @@ message GitHubProviderConfig {
 	string endpoint = 1;
 }
 
+// GitHubAppProviderConfig contains the configuration for the GitHub App provider
+message GitHubAppProviderConfig {
+  // Endpoint is the GitHub API endpoint. If using the public GitHub API, Endpoint can be left blank.
+  string endpoint = 1;
+  // AppName is the GitHub App Name
+  string app_name = 2;
+  // AppID is the GitHub App ID
+  string app_id = 3;
+  // UserID is the GitHub App User ID
+  int64 user_id = 4;
+}
+
 // Context defines the context in which a rule is evaluated.
```

Comment:
I'm unsure if all these fields belong in the provider definition. In the GitHub OAuth definition we only include the `endpoint`, but my assumption is if we're interacting with multiple GitHub endpoints (e.g public vs private), then each one will have it's own GitHub App.

Reply:
So, for the OAuth2 app, we're able to point minder to an enterprise github endpoint. I'm not sure we can do the same with a github app flow? Is that a GitHub cloud-only kind of thing or is this something that may also work in on-prem installations? Would be good to figure out. Else, we can remove the `endpoint` for now and add it when we see the use-case.

Reply:
GitHub Apps are available on GitHub Enterprise Server as well https://docs.github.com/en/apps/sharing-github-apps/making-your-github-app-available-for-github-enterprise-server#each-github-enterprise-server-instance-must-register-their-own-github-app

Reply:
ack, let's include the endpoint then.